components/sections/Navbar/index.tsx

```diff
@@ -0,0 +1,93 @@
+import Image from 'next/image';
+import { useTheme } from 'next-themes';
+import type { FC } from 'react';
+import { useMemo } from 'react';
+
+import LanguageDropdown from '@/components/Common/LanguageDropDown';
+import ThemeToggle from '@/components/Common/ThemeToggle';
+import NavItem from '@/components/sections/NavItem';
+import { useLocale } from '@/hooks/useLocale';
+
+import style from './index.module.css';
+
+type NavItem = { key: string; link: string };
+type NavbarProps = { navItems: NavItem[] };
+
+const Navbar: FC<NavbarProps> = ({ navItems }) => {
+  const { availableLocales, currentLocale } = useLocale();
```

Comment:
As I mentioned locale items should also be a prop.

And you can simply make it:

```tsx
type NavbarProps = { 
  navItems: ComponentProps<NavItem>[];
  languages: Pick<LocaleConfig, 'code' | 'name'>[];
  currentLanguage: string; // just the language code
};
```

Reply:
The `NavItem` component accepts `children` as a prop which is displayed as text. 
so, I should do something like this.

```TS
type NavbarProps = {
  navItems: (Omit<ComponentProps<typeof NavItem>, 'children'> & {
    text: string;
  })[];
  languages: Pick<LocaleConfig, 'code' | 'name'>[];
  currentLanguage: string; // just the language code
  currentTheme: 'dark' | 'light';
  onThemeTogglerClick: () => void; 
```

Reply:
```
navItems: (Omit<ComponentProps<typeof NavItem>, 'children'> & {
    text: string;
  })[];
```

Uh very interesting. NavItem uses children? I wonder if that was intentional per design...

So no, you're right here, we should not do `ComponentProps` in this case, but do the type you've made (`type NavItem = { text, href }`)

Reply:
Here's what the latest NavBarProp looks like:
```TS
type NavbarProps = {
  navItems: NavItem[];
  languages: ComponentProps<typeof LanguageDropdown>;
  onThemeTogglerClick: () => void;
};
```

- The `LanguageDropdown` component requires `currentLanguage` to be of type `{ name, code }`, though it only uses code. So, I used `ComponentProps`.
- `currentTheme` is not required now as it was before used to change theme on the `themeTogglerCallback` but now it is passed as prop.

Also, there's this
> I didn't see the nav items design for mobile. The design hides the nav items and has added a Hambuger icon; how should the items be shown on open state on mobile.

Reply:
> The LanguageDropdown component requires currentLanguage to be of type { name, code }, though it only uses code. So, I used ComponentProps.

I guess you can change the type of LanguageDropdown and make `currentLanguage` to be a string :)

But I still believe you need to add a `currentLanguage` prop, and maybe rename the `languages` to just `availableLanguages`

> currentTheme is not required now as it was before used to change theme on the themeTogglerCallback but now it is passed as prop.

Right, afaik, we don't really need to pass currentTheme, we can simply use Tailwind to render icons with dark/light mode only if needed.

> I didn't see the nav items design for mobile. The design hides the nav items and has added a Hambuger icon; how should the items be shown on open state on mobile.

The hamburger designs are in this page: https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?node-id=696%3A19655&mode=dev

Reply:
Should I use `@radix-ui/react-dropdown-menu` as in the `LanguageDropDown` component for the mobile nav menu? though that would result in a few duplications of markups?

Reply:
> Should I use `@radix-ui/react-dropdown-menu` as in the `LanguageDropDown` component for the mobile nav menu? though that would result in a few duplications of markups?

No. The mobile NavBar is not a dropdown. Let me think about how we should approach this for a little while.

Reply:
Hey, I have added the mobile view. I used CSS sibling selector to show and hide the items but to change the icon from Ham to XMark, I had to use state.

Reply:
Let me review that. I was actually thinking to use the `useMediaQuery` hook to observe on media query changes.

Or simply tailwindcss to mutate the styles, but let me see what you did :)